src/go/rpk/pkg/auth0/auth0.go

```diff
@@ -11,16 +11,15 @@
 package auth0
 
 import (
-	"encoding/json"
+	"context"
 	"errors"
 	"fmt"
-	"io"
 	"net/http"
-	"net/url"
 	"runtime"
 	"time"
 
 	"github.com/lestrrat-go/jwx/jwt"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/httpapi"
 )
 
 // Endpoint groups what url, audience, and clientID to use for getting tokens.
```

Comment:
Let's change ClientID on line 37 below to be a `string`, not `[]string`.

Elsewhere, this PR currently uses clientID[0], which may not exist -- whereas an empty string always will

The intent of supporting multiple client IDs is for ValidateToken, and we can change that API (suggestion in another commit)